go-controller/pkg/ovn/gatewayrouter/policybasedroutes.go

```diff
@@ -0,0 +1,208 @@
+							"error: %v", policy.UUID, nodeName, pbr.clusterRouterName, err)
+					}
+					continue
+				}
+				desiredMatchFound := false
+				for match := range matchTracker {
+					if strings.Contains(policy.Match, match) {
+						desiredMatchFound = true
+						break
+					}
+				}
+				// if the policy is for this node/priority and does not contain a valid match, remove it
+				if !desiredMatchFound {
+					if err := pbr.deletePolicyBasedRoutes(policy.UUID, priority); err != nil {
+						return fmt.Errorf("failed to delete policy route '%s' for host %q on %s "+
+							"error: %v", policy.UUID, nodeName, pbr.clusterRouterName, err)
+					}
+					continue
+				}
+				// now check if the existing policy matches, remove it
+				matchTracker.Delete(policy.Match)
+			}
+		}
+	}
+
+	// cycle through all of the not found match criteria and create new policies
+	for match := range matchTracker {
+		if err := pbr.createPolicyBasedRoutes(match, priority, nexthop); err != nil {
+			return fmt.Errorf("failed to add policy route '%s' for host %q on %s "+
+				"error: %v", match, nodeName, pbr.clusterRouterName, err)
+		}
+	}
+	return nil
+}
+
+func (pbr *PolicyBasedRoutesManager) findPolicyBasedRoutes(priority string) ([]*nbdb.LogicalRouterPolicy, error) {
+	intPriority, _ := strconv.Atoi(priority)
+	networkName := pbr.netInfo.GetNetworkName()
+	p := func(item *nbdb.LogicalRouterPolicy) bool {
+		itemNetworkName, isSecondaryNetwork := item.ExternalIDs[ovntypes.NetworkExternalID]
+		if !isSecondaryNetwork {
+			itemNetworkName = ovntypes.DefaultNetworkName
+		}
+		return itemNetworkName == networkName && item.Priority == intPriority
+	}
+	logicalRouterStaticPolicies, err := libovsdbops.FindLogicalRouterPoliciesWithPredicate(pbr.nbClient, p)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find logical router policy: %v", err)
+	}
+
+	return logicalRouterStaticPolicies, nil
+}
+
+func (pbr *PolicyBasedRoutesManager) deletePolicyBasedRoutes(policyID, priority string) error {
+	lrp := nbdb.LogicalRouterPolicy{UUID: policyID}
+	err := libovsdbops.DeleteLogicalRouterPolicies(pbr.nbClient, pbr.clusterRouterName, &lrp)
+	if err != nil {
+		return fmt.Errorf("error deleting policy %s: %v", policyID, err)
+	}
+
+	return nil
+}
+
+func (pbr *PolicyBasedRoutesManager) createPolicyBasedRoutes(match, priority, nexthops string) error {
+	intPriority, _ := strconv.Atoi(priority)
+	lrp := nbdb.LogicalRouterPolicy{
+		Priority: intPriority,
+		Match:    match,
+		Nexthops: []string{nexthops},
+		Action:   nbdb.LogicalRouterPolicyActionReroute,
+	}
+
+	if pbr.netInfo.GetNetworkName() != ovntypes.DefaultNetworkName {
+		lrp.ExternalIDs = map[string]string{
+			ovntypes.NetworkExternalID:  pbr.netInfo.GetNetworkName(),
+			ovntypes.TopologyExternalID: pbr.netInfo.TopologyType(),
+		}
+	}
+
+	p := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == lrp.Priority && item.Match == lrp.Match
```

Comment:
network scope the predicate

Reply:
ah cool! its fine then but ya, a comment would be great because i expect we will have the same question later.

Reply:
It should be there now.